src/main/java/com/thindeck/dynamo/DyTxn.java

```diff
@@ -74,4 +101,29 @@ public void log(final String text) throws IOException {
     public Iterable<String> log() throws IOException {
         throw new UnsupportedOperationException("#log");
     }
+
+    /**
+     * Method that return the steps array attribute.
+     * @return The array of steps.
+     */
+    public Step[] getSteps() {
+        return this.steps.clone();
+    }
+
+    /**
+     * Method that return the context attribute.
+     * @return The context.
+     */
+    public Context getContext() {
+        return this.context;
+    }
+
+    /**
+     * Method that remove the executed step.
+     */
+    private void removeUsedStep() {
```

Comment:
This method could be removed when `Step` array is converted to a List/Collection